server/src/main/java/io/pravega/schemaregistry/storage/SchemaStore.java

```diff
@@ -0,0 +1,337 @@
+     * group entry from groups table. Deleting a group is idempotent. A group can only be deleted if it is already in Active state. 
+     * Deleting a group is non atomic.
+     * @param namespace namespace
+     * @param group group
+     * @return Completable Future which completes successfully if the group is deleted.  
+     */
+    CompletableFuture<Void> deleteGroup(String namespace, String group);
+
+    /**
+     * Gets an entity tag for the group. All update operations on the group metadata are performed with the entity tag. 
+     * So all metadata operations on group perform compare and set on the metadata and entity tag and if multiple concurrent
+     * updates to group metadata are attempted, they could fail with Write Conflict. 
+     * 
+     * @param namespace namespace
+     * @param group group
+     * @return Completablefuture that holds Entity tag for the group. 
+     */
+    CompletableFuture<Etag> getGroupEtag(String namespace, String group);
+
+    /**
+     * Gets the group properties metadata for the group. This should be an atomic operation. 
+     * 
+     * @param namespace namespace 
+     * @param group group
+     * @return Completable Future which will hold group properties upon completion. 
+     */
+    CompletableFuture<GroupProperties> getGroupProperties(String namespace, String group);
+
+    /**
+     * Updates the compatibility policy for the group conditionally (etag). This should be updated atomically. 
+     * 
+     * @param namespace namespace
+     * @param group group
+     * @param etag entity tag
+     * @param policy new policy 
+     * @return Completable future which is completed successfully if the update completes. 
+     */
+    CompletableFuture<Void> updateCompatibility(String namespace, String group, Etag etag, Compatibility policy);
+
+    /**
+     * Gets latest schemas for all types in the group. The implementation for this API should fetch all latest schemas 
+     * across all types atomically. There is no order on the schemas in the list. 
+     * 
+     * @param namespace namespace
+     * @param group group
+     * @return Completablefuture that holds List of latest schemas with versions for all types in the group. 
+     */
+    CompletableFuture<List<SchemaWithVersion>> getLatestSchemas(String namespace, String group);
+
+    /**
+     * Gets all schemas for all types in the group. The implementation for this API should fetch all schemas 
+     * across atomically. The schema id {@link VersionInfo#id} defines the order in which the schemas were added and the list
+     *      * is ordered by the schema id. 
+     *
+     * @param namespace namespace
+     * @param group group
+     * @return Completablefuture that holds List of latest schemas with versions for all types in the group. 
+     */
+    CompletableFuture<List<SchemaWithVersion>> listSchemas(String namespace, String group);
```

Comment:
Looks very similar to `getLatestSchemas` with the difference that this one fetches all and that one fetches the latest ones only. The names of these two methods should be similar as well. Or may you could have a single method with a boolean parameterization.

Reply:
actually their business logic is quite different. So using same method would be unwise. 
Keeping two separate methods is cleaner that way.